pkg/asset-manager-utils/contracts/IAssetManager.sol

```diff
@@ -48,6 +48,11 @@ interface IAssetManager {
      */
     function updateBalanceOfPool(bytes32 poolId) external;
 
+    /**
+     * @notice Checks whether the pool should rebalance upon adding or removing `delta` cash.
+     */
+    function shouldRebalance(uint256 delta, bool positiveDelta) external view returns (bool);
```

Comment:
I'm thinking about how this will be called by the Pool. We either a) store the list of asset managers in the Pool itself (a bit cumbersome as we'll need to do the whole if-else chain for `immutable` pseudo array lookup), or b) query the Vault for the AM.

Interestingly, if we were to query the AM, we'd be making the exact same call the AM would make: `vault.getPoolTokenInfo`, which returns `cash`, `managed` and `assetManager`. So I'm thinking we could do the following:
 - call `getPoolTokenInfo` from the Pool to query the AM
 - if the AM is non-zero, call `shouldRebalance(cash +- delta, managed)`, meaning the AM would skip the Vault query entirely and just go 'given this cash and managed, should I rebalance?'

This would also avoid the not-so-great boolean in the interface. What do you think?

Reply:
I think that's much cleaner and could be used to simplify our tests too.